src/backend/distributed/planner/extended_op_node_utils.c

```diff
@@ -42,15 +46,25 @@ static bool ShouldPullDistinctColumn(bool repartitionSubquery,
  * value should be used in a read-only manner.
  */
 ExtendedOpNodeProperties
-BuildExtendedOpNodeProperties(MultiExtendedOp *extendedOpNode, bool
-							  pullUpIntermediateRows)
+BuildExtendedOpNodeProperties(MultiExtendedOp *extendedOpNode,
+							  bool pullUpIntermediateRows)
 {
 	ExtendedOpNodeProperties extendedOpNodeProperties;
 
 	List *tableNodeList = FindNodesOfType((MultiNode *) extendedOpNode, T_MultiTable);
 	bool groupedByDisjointPartitionColumn =
+		!extendedOpNode->hasNonPushableWindowFunction &&
```

Comment:
This may be a bit heavy handed, but essentially a lot of logic assumes that if this is true we can push down grouping, but when a window function can't be pushed down we still need to group it on the coordinator

Reply:
I meant that we assume we can push down mostly everything, but we can't push down non pushable window functions